tests/test_fix_course_delete_module_class_reporter_test.php

```diff
@@ -0,0 +1,244 @@
+<?php
+// This file is part of Moodle - https://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <https://www.gnu.org/licenses/>.
+
+namespace tool_fix_delete_modules;
+
+use Exception;
+use tool_fix_delete_modules\reporter;
+
+defined('MOODLE_INTERNAL') || die();
+require_once(__DIR__ . "/../classes/diagnosis.php");
+require_once(__DIR__ . "/../classes/deletemodule.php");
+require_once(__DIR__ . "/../classes/deletetasklist.php");
+
+/**
+ * The test_fix_course_delete_module_class_reporter test class.
+ *
+ * Tests for the reporter class.
+ *
+ * @package     tool_fix_delete_modules
+ * @category    test
+ * @author      Brad Pasley <brad.pasley@catalyst-au.net>
+ * @copyright   Catalyst IT, 2022
+ * @license     https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class test_fix_course_delete_module_class_reporter_test extends \advanced_testcase {
```

Comment:
It seems like `get_tables_report()` is a critical public method but it's not covered by tests